zero/bc_basic.c

```diff
@@ -125,20 +125,20 @@ gkyl_bc_basic_advance(const struct gkyl_bc_basic *up, struct gkyl_array *buff_ar
     case GKYL_BC_PKPM_MOM_REFLECT:
     case GKYL_BC_PKPM_MOM_NO_SLIP:
       gkyl_array_copy_to_buffer_fn(buff_arr->data, f_arr,
-                                   *up->skin_r, up->array_copy_func->on_dev);
+                                   up->skin_r, up->array_copy_func->on_dev);
       break;
 
     case GKYL_BC_REFLECT:
     case GKYL_BC_PKPM_SPECIES_REFLECT:
       gkyl_array_flip_copy_to_buffer_fn(buff_arr->data, f_arr, up->dir+up->cdim,
-                                        *up->skin_r, up->array_copy_func->on_dev);
+                                        up->skin_r, up->array_copy_func->on_dev);
       break;
 
     case GKYL_BC_FIXED_FUNC: // if BC is fixed func, do nothing, buffer already full
       break;
   }
   // 2) Copy from buffer to ghost.
-  gkyl_array_copy_from_buffer(f_arr, buff_arr->data, *up->ghost_r);
+  gkyl_array_copy_from_buffer(f_arr, buff_arr->data, &(*up->ghost_r));
```

Comment:
Not sure what's going on here, but if you have & and * you can just delete both of them

Reply:
@akashukla yeah this is not good. please remove &(* throughout